src/main/scala/com/amazon/deequ/comparison/DataSynchronization.scala

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may not
+ * use this file except in compliance with the License. A copy of the License
+ * is located at
+ *
+ *     http://aws.amazon.com/apache2.0/
+ *
+ * or in the "license" file accompanying this file. This file is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ */
+
+package com.amazon.deequ.comparison
+
+import org.apache.spark.sql.{Column, DataFrame}
+import org.apache.spark.sql.functions.col
+
+object DataSynchronization {
+
+  /**
+   * Compare two DataFrames 1 to 1 with specific columns inputted by the customer.
+   *
+   * @param ds1       The first data set which the customer will select for comparison.
+   * @param ds2       The second data set which the customer will select for comparison.
+   * @param colKeyMap A map of columns to columns used for joining the two datasets.
+   *                  The keys in the map are composite key forming columns from the first dataset.
+   *                  The values for each key is the equivalent column from the second dataset. 
+   * @param compCols  A map of columns to columns which we will check for equality, post joining.
+   *                  This is an optional parameter. If not provided, we assume that the non primary
+   *                  key columns in both datasets are identical. If provided, we expect a map like colKeyMap.
+   * @param assertion A function which accepts the match ratio and returns a Boolean.
+   * @return Boolean  Internally we calculate the ratio of the rows that match and we run the assertion
+   *                  on that outcome that ends up being a true or false response.
+   */
+
+  def columnMatch(ds1: DataFrame,
+                  ds2: DataFrame,
+                  colKeyMap: Map[String, String],
+                  compCols: Option[Map[String, String]],
```

Comment:
this can default to an empty map instead of `Option`

Reply:
Added overloaded methods.